### ISSUE TITLE
GetTensor MagnetoDynamicsUtils.F90 leaks memory

### ISSUE THREAD

**Issue by juhanikataja** [ElmerCSC/elmerfem#130]
There is a work variable Cwrk set to NULL() which is passed to ListGetRealArray which allocates corresponding variable and no-one deallocates it. Anyways, it seems bad idea to allocate the work array at each element.

Snip:
```fortran
! GetTensor line 154
    Cwrk => NULL()
! Lists.F90 line 5252
    IF ( .NOT.ASSOCIATED( F ) ) THEN
       ALLOCATE( F(N1,N2,N) )
     ELSE IF ( SIZE(F,1)/=N1.OR.SIZE(F,2)/=N2.OR.SIZE(F,3)/= N ) THEN
       DEALLOCATE( F )
       ALLOCATE( F(N1,N2,N) )
     END IF
```
Thanks to @jvencels for finding the leak.

**Comment by juhanikataja** [comment]
The amount of memory leaked is proportional to timesteps taken.

**Comment by juharu** [comment]
I think the initialization should be moved to the definition of the variable and removed from 
the executable code. That should do it. 
Br, Juha

**Comment by mbycklin** [comment]
Initialization to avoid implicit SAVE was added in commit e9fd3936771fd5aea79657fdc6e3110dc8f4fc74 to fix a race condition. If local work storage is needed **Cwrk** should be defined private to each OpenMP thread  to avoid data races. That is, in addition to
`REAL(KIND=dp), SAVE, POINTER :: Cwrk(:,:,:) => NULL()`
the definition 
`!$OMP THREADPRIVATE(CWrk)`
should be added to the end of the function definition section.

**Comment by mbycklin** [comment]
A correction to this issue and a few others which were similar in https://github.com/ElmerCSC/elmerfem/pull/128/commits/0aad3dba419dd4f44dbbcb1f9c26fc6d8a657c50.

**Comment by mbycklin** [comment]
Corrected in merged pull request #128.
